### PR TITLE
Moved edit to cover select and update manga info view

### PR DESCRIPTION
**Pull request by NoodleMage**
Feedback is welcomed :).

### CODE REVIEW

**Comment by NoodleMage** [comment]
screenshot_20160214-153131

**Comment by inorichi** [comment]
It's looking good :+1:

**Comment by j2ghz** [comment]
Could the icon be changed? Everytime I see the heart I think *add to favourites* or *like*, not *add to library*.

Maybe something like:
http://www.endlessicons.com/wp-content/uploads/2014/03/bookmark-icon-2-614x460.png
http://www.endlessicons.com/wp-content/uploads/2014/03/bookmark-icon-3-614x460.png
https://image.freepik.com/free-icon/book-with-white-bookmark_318-49515.jpg

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryCategoryFragment.java [hunk 1]
```diff
@@ -47,6 +48,8 @@ public static LibraryCategoryFragment newInstance(int position) {
         return fragment;
     }
 
+    LibraryFragment f;
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryCategoryFragment.java]: Try not to hold a global reference to fragments/activities

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryCategoryAdapter.java [hunk 2]
```diff
@@ -3,8 +3,10 @@
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.ImageView;
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryCategoryAdapter.java]: File > Settings > Editor > General > Auto import > Optimize imports on the fly.

No changes needed on this class. :P

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java [hunk 3]
```diff
@@ -213,6 +240,15 @@ public boolean onActionItemClicked(ActionMode mode, MenuItem item) {
         return false;
     }
 
+    /**
+     * TODO workaround. Covers won't refresh any other way.
+     */
+    public void rebuildAdapter()
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java]: You can try calling only `viewPager.setAdapter(adapter)`. I'm not sure if it works though.

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoFragment.java [hunk 4]
```diff
@@ -112,14 +96,33 @@ private void setMangaInfo(Manga manga, Source mangaSource) {
...
 
     public void setChapterCount(int count) {
         chapterCount.setText(String.valueOf(count));
     }
 
-    private void setFavoriteText(boolean isFavorite) {
-        favoriteBtn.setText(!isFavorite ? R.string.add_to_library : R.string.remove_from_library);
+    private void setFavoriteDrawable(boolean isFavorite) {
+        if (isFavorite) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoFragment.java]: Probably with `ContextCompat` there's no need to check SDK version

**Comment by M2ck** [comment]
Agree with @j2ghz about "Add to library" icon, I suggest this one: https://materialdesignicons.com/icon/library-plus

**Comment by NoodleMage** [comment]
Thanks for feedback :+1: 

@M2ck Problem with that one is that you can't see if it is added or not because it has no border version.
I myself prefer https://design.google.com/icons/#ic_book  (same as toolbar) but has the same problem.

@j2ghz  I think that one is better yes. Will change it (also found out that I forgot to upload the drawables :O.

**Comment by j2ghz** [comment]
could you change it in catalogue as well?

**Comment by NoodleMage** [comment]
@j2ghz  Manga info is one fragment that is called from Library and from Catalogue. So it is changed for both.

**Comment by j2ghz** [comment]
I mean in catalogue there is a blue heart on manga that are in library

**Comment by NoodleMage** [comment]
Sorry didn't understand you :).

@inorichi  May I change it?

**Comment by NoodleMage** [comment]
@j2ghz  @M2ck

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java (hunk 3 above)
Comment (NoodleMage) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java]: Doesn't work :(.

**Comment by inorichi** [comment]
Yes, I'm fine with it.

**Comment by M2ck** [comment]
@NoodleMage :+1:

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java [hunk 5]
```diff
@@ -187,6 +192,16 @@ public void setContextTitle(int count) {
         actionMode.setTitle(getString(R.string.label_selected, count));
     }
 
+    public void setVisibilityOfCoverEdit(int count)
+    {
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java]: Please don't use brackets on new lines D:

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java [hunk 6]
```diff
@@ -42,16 +48,15 @@
 public class LibraryFragment extends BaseRxFragment<LibraryPresenter>
         implements ActionMode.Callback {
 
+    private static final int REQUEST_IMAGE_OPEN = 101;
+    protected LibraryAdapter adapter;
     @Bind(R.id.view_pager) ViewPager viewPager;
+    @State int activeCategory;
+    @State String query = "";
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java]: In my opinion placing every variable together is worse for readability. I separate them by some kind of meaning (bindings in a group, similar properties in another group, etc...)

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java (hunk 3 above)
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java]: Try implementing `getItemPosition` in `LibraryAdapter` and returning `POSITION_NONE` there.

File: app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java [hunk 7]
```diff
@@ -226,6 +255,55 @@ public void destroyActionModeIfNeeded() {
         }
     }
 
+    private void changeSelectedCover(List<Manga> mangas) {
+        if (!(mangas.size() > 1))
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/library/LibraryFragment.java]: `mangas.size() == 1`? It also avoids empty list. And bracket on the same line :(

File: app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoFragment.java [hunk 8]
```diff
@@ -112,14 +96,25 @@ private void setMangaInfo(Manga manga, Source mangaSource) {
...
     }
 
-    private void setFavoriteText(boolean isFavorite) {
-        favoriteBtn.setText(!isFavorite ? R.string.add_to_library : R.string.remove_from_library);
+    private void setFavoriteDrawable(boolean isFavorite) {
+        if (isFavorite) {
+            fabFavorite.setImageDrawable(ContextCompat.getDrawable(getContext(), R.drawable.ic_bookmark_white_24dp));
+        } else {
+            fabFavorite.setImageDrawable(ContextCompat.getDrawable(getContext(), R.drawable.ic_bookmark_border_white_24dp));
+        }
```
Comment (inorichi) [inline app/src/main/java/eu/kanade/tachiyomi/ui/manga/info/MangaInfoFragment.java]: fabFavorite.setImageDrawable(ContextCompat.getDrawable(getContext(), isFavorite ?   
        R.drawable.ic_bookmark_white_24dp : 
        R.drawable.ic_bookmark_border_white_24dp));

This looks better IMO.

**Comment by inorichi** [comment]
Is it ready to merge @NoodleMage ? Also, please next time try not to include files with only line breaks in the commit (no need to fix it this time) :+1:

**Comment by NoodleMage** [comment]
You can merge it :). Will try to prevent it next time ;-(

**Pull request merged**